log/root.go

```diff
@@ -2,42 +2,30 @@ package log
 
 import (
 	"os"
-	"sync"
 
 	"golang.org/x/exp/slog"
 )
 
-var (
-	rootMu sync.Mutex
-	root logger
-)
+var root Logger
 
 func init() {
 	defaultLogger := &logger{slog.New(DiscardHandler())}
 	SetDefault(defaultLogger)
 }
 
+// SetDefault sets the default global logger
 func SetDefault(l Logger) {
-	rootMu.Lock()
-	defer rootMu.Unlock()
-	root := l
+	root = l
```

Comment:
I would go with atomic, like you did before, and like slog does https://cs.opensource.google/go/x/exp/+/master:slog/logger.go;l=30